R/generic_functions.R

```diff
@@ -0,0 +1,13 @@
+#' @title Generic Function for Extracting Datasets
+#'
+#' @description This function serves as a generic method to extract datasets
+#' from objects of various classes. It dispatches to specific methods based
+#' on the class of the input object.
+#'
+#' @param object An object from which to extract the dataset.
+#' @param ... Additional arguments passed to other functions.
+#' @return The get_dataset extracted from the object.
+#' @keywords internal
+get_dataset <- function(object, ...) {
```

Comment:
What roles does the `get_dataset()` function play, I can see that it is internal, and is not used in any functions in the package. One of the methods, `get_dataset.match()`is used in tests but not in functions. Could the generic and methods be removed from the package?

Reply:
This was a function introduced in a previous release, but right now it lacks use. I will remove it as part of issue https://github.com/epiverse-trace/vaccineff/issues/78